edge/client/portainer_edge_client.go

```diff
@@ -51,10 +51,6 @@ type NonOkResponseError struct {
 	msg string
 }
 
-func newNonOkResponseError(msg string) *NonOkResponseError {
-	return &NonOkResponseError{msg: msg}
-}
-
```

Comment:
You cannot remove this, we rely on that error type in other parts of the code.

Reply:
The type was not removed, only the constructor function was removed, because I found that it is not used.

Reply:
That is technically true but the constructor function is not used anymore because you removed it in this PR. :rofl: 

You can't do that, we need to return that error because we depend on it in other places, please see: https://github.com/portainer/agent/blob/676d62a86b33071fe51172c42fca74c80df3ac95/edge/poll.go#L217-L225